revocation/proof.go

```diff
@@ -371,12 +374,12 @@ func (p *proof) ProofResult(name string) *big.Int {
 }
 
 func (p *proof) verify(pk *PublicKey) bool {
-	grp := (*qrGroup)(pk.Group)
+	grp := (*prooftools.PublicKeyGroup)(pk.Group)
 	commitments := proofstructure.commitmentsFromProof(grp, []*big.Int{}, p.Challenge, grp, p, p)
 	return (*Proof)(p).VerifyWithChallenge(pk, common.HashCommit(commitments, false))
 }
 
-func (s *proofStructure) commitmentsFromSecrets(g *qrGroup, list []*big.Int, bases keyproof.BaseLookup, secretdata keyproof.SecretLookup) ([]*big.Int, proofCommit) {
+func (s *proofStructure) commitmentsFromSecrets(g *prooftools.PublicKeyGroup, list []*big.Int, bases keyproof.BaseLookup, secretdata keyproof.SecretLookup) ([]*big.Int, proofCommit) {
```

Comment:
Here too, and in `commitmentsFromProofs`, I think it might be easier to accept a `*keys.PublicKey` instead.

Reply:
fixed